### PR TITLE
Improve toHash function

### PR DESCRIPTION
**Pull request by Mogztter**

### CODE REVIEW

File: src/asciidoctor-core-api.js [hunk 1]
```diff
@@ -3,7 +3,7 @@
  * @private
  */
 var toHash = function (object) {
-  return object && !('$$smap' in object) ? Opal.hash(object) : object;
+  return !object || (object && object.$$is_hash) ? object : Opal.hash2(Object.keys(object), object);
```
Comment (mojavelinux) [inline src/asciidoctor-core-api.js]: I don't think the first negative check is needed. I'd simply write it as:

```js
object && !object.$$is_hash ? Opal.hash2(Object.keys(object), object) : object;
```

Comment (Mogztter) [reply src/asciidoctor-core-api.js]: Yes you're right, I don't know why I wrote this complicated test... :thinking:

Comment (mojavelinux) [reply src/asciidoctor-core-api.js]: :tada:

**Comment by Mogztter** [comment]
@mojavelinux Can we merge ?

**Comment by mojavelinux** [comment]
Yep!

**Pull request merged**